### ISSUE TITLE
Level names in multi-level index are ignored

### ISSUE THREAD

**Issue by MaksimZh** [pydata/xarray#5646]
**What happened**:
The difference of the level names in multi-level index is silently ignored.

**What you expected to happen**:
Error message.

**Minimal Complete Verifiable Example**:

```python
import xarray as xr

a = xr.DataArray([[1]], dims = ["x", "y"])

b = a.stack({"z": ["x", "y"]})
c = a.rename({"x": "not-x", "y": "not-y"}).stack({"z": ["not-x", "not-y"]})

xr.testing.assert_allclose(b, c)
print(b == c, "\n")
print(b.coords == c.coords, "\n")
print(b + c, "\n")
print(c + b, "\n")
```

Output:
```
<xarray.DataArray (z: 1)>
array([ True])
Coordinates:
  * z        (z) MultiIndex
  - x        (z) int64 0
  - y        (z) int64 0

True

<xarray.DataArray (z: 1)>
array([2])
Coordinates:
  * z        (z) MultiIndex
  - x        (z) int64 0
  - y        (z) int64 0

<xarray.DataArray (z: 1)>
array([2])
Coordinates:
  * z        (z) MultiIndex
  - not-x    (z) int64 0
  - not-y    (z) int64 0

```

**Environment**:

<details><summary>Output of <tt>xr.show_versions()</tt></summary>

INSTALLED VERSIONS
------------------
commit: None
python: 3.9.1 (tags/v3.9.1:1e5d33e, Dec  7 2020, 17:08:21) [MSC v.1927 64 bit 
(AMD64)]
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 165 Stepping 5, GenuineIntel
byteorder: little
LC_ALL: None
LANG: ru_RU.UTF-8
LOCALE: ('Russian_Russia', '1251')
libhdf5: 1.12.0
libnetcdf: None

xarray: 0.18.2
pandas: 1.3.0
numpy: 1.21.1
scipy: 1.6.0
netCDF4: None
pydap: None
h5netcdf: None
h5py: 3.1.0
Nio: None
zarr: None
cftime: None
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: None
dask: None
distributed: None
matplotlib: 3.4.2
cartopy: None
seaborn: None
numbagg: None
pint: None
setuptools: 57.4.0
pip: 21.1.3
conda: None
pytest: None
IPython: None
sphinx: None

</details>

**Comment by MaksimZh** [comment]
Note that comparison of pandas.MultiIndex instances ignores the names.
This may be the cause of the issue.

```python
import pandas as pd

a = pd.MultiIndex.from_tuples([(0, 0)], names=["x", "y"])
b = pd.MultiIndex.from_tuples([(0, 0)], names=["not-x", "not-y"])

print(a == b)
print(a.names == b.names)
```

Output:
```
[ True]
False
```

**Comment by benbovy** [comment]
Thanks for the report @MaksimZh!

With the ongoing index refactoring (see https://github.com/pydata/xarray/projects/1), we will be able to fix that:

- we plan to drop multi-index "virtual" coordinates and expose levels as "real" coordinates instead
- pandas indexes are being wrapped in `xarray.Index` subclasses, so we'll be able to check for the level names when comparing indexes

**Comment by benbovy** [comment]
This is fixed in v2022.6.0

```python
xr.testing.assert_allclose(b, c)
# AssertionError: Left and right DataArray objects are not close
# 
# Coordinates only on the left object:
#   * x        (z) int64 0
#   * y        (z) int64 0
# Coordinates only on the right object:
#   * not-y    (z) int64 0
#   * not-x    (z) int64 0

print(b == c, "\n")
# ValueError: cannot re-index or align objects with conflicting indexes found for the following coordinates: 'z' (2 conflicting indexes)
# Conflicting indexes may occur when
# - they relate to different sets of coordinate and/or dimension names
# - they don't have the same type
# - they may be used to reindex data along common dimensions


```
